extensions/mongodb-client/deployment/src/main/java/io/quarkus/mongodb/deployment/MongoClientProcessor.java

```diff
@@ -66,6 +72,20 @@
     private static final DotName MONGO_CLIENT = DotName.createSimple(MongoClient.class.getName());
     private static final DotName REACTIVE_MONGO_CLIENT = DotName.createSimple(ReactiveMongoClient.class.getName());
 
+    static class MongoClientTracingEnabled implements BooleanSupplier {
+        MongoClientBuildTimeConfig mConfig;
+
+        @Override
+        public boolean getAsBoolean() {
+            return mConfig.tracingEnabled;
+        }
+    }
+
+    @BuildStep(onlyIf = MongoClientTracingEnabled.class)
+    IndexDependencyBuildItem addMongoTracingDependencies() {
+        return new IndexDependencyBuildItem("io.opentracing.contrib", "opentracing-mongo-common");
+    }
+
```

Comment:
Don't index it if you don't want to include it automatically later !

Reply:
Removed